third_party/move/move-compiler-v2/tests/ability-check/v1-typing/phantom_param_struct_decl_invalid.exp

```diff
@@ -0,0 +1,50 @@
+
+Diagnostics:
+error: phantom type `T` can only be used as an argument for another phantom type parameter
+  ┌─ tests/ability-check/v1-typing/phantom_param_struct_decl_invalid.move:4:12
+  │
+4 │         a: T,
+  │            ^
+  │
+  = required by declaration of field `a`
+
+error: phantom type `T` can only be used as an argument for another phantom type parameter
+   ┌─ tests/ability-check/v1-typing/phantom_param_struct_decl_invalid.move:12:15
+   │
+10 │     struct S2<T> { a: T }
+   │               - declaration of type parameter `T`
+11 │     struct S3<phantom T> {
+12 │         a: S2<T>
+   │               ^
+   │
+   = required by instantiating type parameter `T` of struct `S2`
```

Comment:
This error message is much worse than the one in V1:
```
error[E02013]: invalid phantom type parameter usage
   ┌─ tests/move_check/typing/phantom_param_struct_decl_invalid.move:12:15
   │
11 │     struct S3<phantom T> {
   │                       - 'T' declared here as phantom
12 │         a: S2<T>
   │               ^ Phantom type parameter cannot be used as an argument to a non-phantom parameter
```
We should add a similar message ("Phantom type parameter cannot be used as an argument to a non-phantom parameter") to the label on line 12.

Please file an issue.